perma_web/perma/reporter.py

```diff
@@ -0,0 +1,115 @@
+from copy import copy
+
+from django.views import debug
+from django.utils import log
+from django.conf import settings
+from django import template
+
+
+#
+# Templates are embedded in the file so that we know the error handler will
+# always work even if the template loader is broken.
+#
+
+
```

Comment:
Is this the whole usual template, minus the settings, or are there other changes?

Reply:
I copied it from somewhere like https://github.com/django/django/blob/90db4bb0d72e4731052bd33500840fff00834768/django/views/debug.py#L1059-L1140 -- sorry I don't have a nice diff. Trying to make one.... here it is:

```
$ diff 500.orig.txt 500.new.txt 
10d9
< Python Path: {{ sys_path }}
12,16d10
< Installed Applications:
< {{ settings.INSTALLED_APPS|pprint }}
< Installed Middleware:
< {{ settings.MIDDLEWARE|pprint }}"""
< """
55,57c49
< {% if user_str %}USER: {{ user_str }}{% endif %}
< 
< GET:{% for k, v in request_GET_items %}
---
> GET:{% for k, v in request.GET.items %}
60c52
< POST:{% for k, v in filtered_POST_items %}
---
> POST:{% for k, v in filtered_POST.items %}
63c55
< FILES:{% for k, v in request_FILES_items %}
---
> FILES:{% for k, v in request.FILES.items %}
65,71d56
< 
< COOKIES:{% for k, v in request_COOKIES_items %}
< {{ k }} = {{ v|stringformat:"r" }}{% empty %} No cookie data{% endfor %}
< 
< META:{% for k, v in request.META.items|dictsort:0 %}
< {{ k }} = {{ v|stringformat:"r" }}{% endfor %}
< {% else %}Request data not supplied
73,75d57
< Settings:
< Using settings module {{ settings.SETTINGS_MODULE }}{% for k, v in settings.items|dictsort:0 %}
< {{ k }} = {{ v|stringformat:"r" }}{% endfor %}
82c64
< """)  # NOQA
---
> """)
```

Reply:
Ah, thanks, that's super helpful.